src/gui/src/staGui.cpp

```diff
@@ -641,15 +641,16 @@ void TimingPathRenderer::highlightNode(const TimingPathNode* node, TimingPath::T
       auto node_itr = std::find_if(nodes->begin(), nodes->end(), [node](const auto& other) -> bool {
         return other.get() == node;
       });
+      int node_index = std::distance(nodes->begin(), node_itr);
 
-      if ((node_itr + 1) != nodes->end()) {
+      if ((node_index + 1) != nodes->size()) {
```

Comment:
This doesn't look right - I think you need 

if ((node_index + 1) < nodes->size()) {

if the node isn't found, node_index == nodes->size() and you go awry.

I think this will also fail if nodes->empty() (there is no valid node_index).  I'm not sure if that can happen.

Reply:
the node list should never be empty at this point, but I added a check anyways (just in case).  I think there is a lot of duplicated code in this file and at some point I'd like to go over it and clean it up (just not at time time).